### ISSUE TITLE
Web3.js 1.2.5 tx sending broken in v8?

### ISSUE THREAD

**Issue by danfinlay** [MetaMask/metamask-extension#8926]
User on reddit reports txs sent with web3.js on Mmv8 return promises that never resolve, even on ropsten.
https://www.reddit.com/r/ethdev/comments/hlp4i6/issues_with_metamaks_v8_on_firefox/?utm_source=share&utm_medium=ios_app&utm_name=iossmf

Ref: #8894

**Comment by rekmarks** [comment]
Reproduced with `web3@1.2.5`.

**Comment by rekmarks** [comment]
See updated previous comment; I cannot reproduce, and am beginning to suspect we've fixed it.

**Comment by hbarcelos** [comment]
I'm the user from Reddit who reported the problem.

More context:
1. This happens on Kovan on Firefox with MetaMask 8.x.
2. Chrome still running 7.x, so everything works as it should.

A few days ago I reported yet another bug of Kovan being reported as having chain ID `66` instead of `42` (https://twitter.com/hbarcelos909/status/1279071557214748673) which @rekmarks fixed, but maybe it's related.

I'm running v8.0.2:
![MetaMask Version](https://user-images.githubusercontent.com/16565602/86604576-ef022800-bf7b-11ea-94d1-a8cb47769c2a.png)

It happens with any transaction where I interact with a Smart Contract.

To give you a concrete example, my code is exaclty this one:

```javascript
async function submitTranslation({ ID, text }, { from, gas, gasPrice } = {}) {
  const tx = linguo.methods.submitTranslation(ID, text).send({
    from,
    gas,
    gasPrice,
  });

  return { tx };
}
```

Specifically for this call, `gas` and `gasPrice` are `undefined`.

`tx` above is the infamous `PromiEvent` from `web3.js`. The promise is never fulfilled or rejected, neither the `receipt`, `confirmation` or `error` events are ever emitted.

Here's the source code for the contract instance in `linguo` variable:
https://github.com/kleros/kleros-interaction/blob/master/contracts/standard/arbitration/Linguo.sol#L270

I included some `console.debug` statements where I can see the following:

![debug](https://user-images.githubusercontent.com/16565602/86605648-59679800-bf7d-11ea-8d29-7651f138193f.png)

The timestamps are in UTC-3.

Notice now that the transaction was actually mined a few seconds later:
https://kovan.etherscan.io/tx/0x421e98f3c12230499962a73c58bf0ad368ca82d59c4a220b1c67a0c9e620cf07

![txn](https://user-images.githubusercontent.com/16565602/86605857-9df33380-bf7d-11ea-9822-632a25dd1ca5.png)

I'm here 20+ minutes later with a reference to `tx` and if I `console.log` it, I still get:

![console.log](https://user-images.githubusercontent.com/16565602/86606012-d09d2c00-bf7d-11ea-9a37-ccf23fe47ccd.png)

**Comment by whymarrh** [comment]
@hbarcelos would you be able to share a minimal complete example? Maybe a https://codesandbox.io or JS Bin?

**Comment by Gudahtt** [comment]
I was able to reproduce the problem using web3@1.2.5 and MetaMask v8.0.2. It has been fixed on `develop` though. I'm currently confirming which commit is responsible for this being fixed.

**Comment by Gudahtt** [comment]
I've just confirmed that this was fixed by #8921. This will be included in v8.0.3, which we're hoping to publish sometime today.
